scripts/check_formatting

```diff
@@ -7,10 +7,27 @@ readonly SCRIPTS_DIR="$(dirname "$(readlink -f "$0")")"
 source "$SCRIPTS_DIR/common"
 
 # Check which BUILD and .bzl files need to be reformatted.
-find oak examples toolchain \( -type f -name BUILD -or -name WORKSPACE -or -name '*bzl' \) \
-  -exec buildifier -lint=warn -mode=check {} +
-find oak examples toolchain \( -type f -name BUILD \) \
-  -exec grep --files-without-match '^    licenses = \["notice"\],$' {} + && (echo 'missing license directive in BUILD file'; exit 1)
+# TODO: Refactor the file matching logic into a separate bash function, or use a real programming
+# language for the job.
+find . \
+  \(  \
+    -not \( -path ./bazel-cache -prune \) -and \
+    -not \( -path ./cargo-cache -prune \) -and \
+    -not \( -path ./examples/target -prune \) -and \
+    -not \( -path ./sdk/rust/target -prune \) -and \
+    -not \( -path ./third_party -prune \) \
+    \( -type f -name BUILD -or -name WORKSPACE -or -name '*bzl' \) \
+  \) -exec buildifier -lint=warn -mode=check {} +
+
+find . \
+  \(  \
+    -not \( -path ./bazel-cache -prune \) -and \
+    -not \( -path ./cargo-cache -prune \) -and \
+    -not \( -path ./examples/target -prune \) -and \
+    -not \( -path ./sdk/rust/target -prune \) -and \
+    -not \( -path ./third_party -prune \) \
+    \( -type f -name BUILD \) \
+  \) -exec grep --files-without-match '^    licenses = \["notice"\],$' {} + && (echo 'missing license directive in BUILD file'; exit 1)
 
 # Check shell scripts for common mistakes.
 find scripts -type f -exec shellcheck {} +
```

Comment:
pre-existing nit: remove "and rustfmt" from the comment below here -- the rustfmt step now has its own comment.

Reply:
Done.